### ISSUE TITLE
User is disabled. (Service: AWSCognitoIdentityProviderService; Status Code: 400; Error Code: NotAuthorizedException;)(aws-amplify-react-native)

### ISSUE THREAD

**Issue by nihp** [aws-amplify/amplify-js#5503]
I got an error as an internal server error in my UI. While checking the Cognito it shows for User is disabled. NotAuthorizedException.

In which case the user is disabled and how can I know the user is disabled for the end-user in aws-amplify-react-native

**Comment by harrysolovay** [comment]
This is the expected behavior; it would be insecure for the Cognito service to show that users exist but are disabled. Bad actors could check to see whether an email account / phone number / etc. is registered.

If this is essential to your use case, you can create an endpoint for using the admin APIs to check the disabled status of a user. But this is not something that we would support through the front-end library. And it's likely not something that the CLI team would add to their roadmap (this is an uncommon use case).

In summary, you could [create an endpoint](https://docs.amplify.aws/cli/function) that accepts the attempt login info and sends an [AdminGetUser](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminGetUser.html) request. The response will contain a [UserStatus](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminGetUser.html#API_AdminGetUser_ResponseSyntax) property, which tells you whether the user is disabled.
